include/seqan3/range/view/pairwise_combine.hpp

```diff
@@ -0,0 +1,618 @@
+ *
+ * ### View properties
+ *
+ * | range concepts and reference_t  | `urng_t` (source range type)          | `rrng_t` (returned range type)                                       |
+ * |---------------------------------|:-------------------------------------:|:--------------------------------------------------------------------:|
+ * | std::ranges::InputRange         | *required*                            | *preserved*                                                          |
+ * | std::ranges::ForwardRange       | *required*                            | *preserved*                                                          |
+ * | std::ranges::BidirectionalRange |                                       | *preserved*                                                          |
+ * | std::ranges::RandomAccessRange  |                                       | *preserved*                                                          |
+ * | std::ranges::ContiguousRange    |                                       | *lost*                                                               |
+ * |                                 |                                       |                                                                      |
+ * | std::ranges::ViewableRange      | *required*                            | *guaranteed*                                                         |
+ * | std::ranges::View               |                                       | *guaranteed*                                                         |
+ * | std::ranges::SizedRange         |                                       | *preserved*                                                          |
+ * | std::ranges::CommonRange        | *required*                            | *guaranteed*                                                         |
+ * | std::ranges::OutputRange        |                                       | *lost*                                                               |
+ * | seqan3::const_iterable_concept  |                                       | *preserved*                                                          |
+ * |                                 |                                       |                                                                      |
+ * | seqan3::reference_t             |                                       | std::tuple<seqan3::reference_t<urng_t>, seqan3::reference_t<urng_t>> |
+ *
+ * See the \link view view submodule documentation \endlink for detailed descriptions of the view properties.
+ *
+ * ### Thread safety
+ *
+ * Concurrent access to this view, e.g. while iterating over it, is thread-safe and must not be protected externally.
+ *
+ * ### Example
+ *
+ * \include test/snippet/range/view/pairwise_combine.cpp
+ *
+ * \attention This view cannot be chained immediately with an infinite range, because upon construction it will
+ *            take forever to reach the last element of the view.
```

Comment:
Seeing all the difficulties we already have with specifiying the behaviour of this view, can we create a TODO-card to remove the requirement for CommonRange and the inability to handle infinite ranges?

Reply:
I'll make a card for it.